### PR TITLE
CHE-1118. Correct handling of search results when the occurrences more 1000

### PR DESCRIPTION
**Pull request by RomanNikitenko**
@vparfonov 

Signed-off-by: Roman Nikitenko rnikitenko@codenvy.com

### CODE REVIEW

**Comment by RomanNikitenko** [comment]
@vparfonov @skabashnyuk I've updated pull request, please review

File: wsagent/che-core-api-project/src/main/java/org/eclipse/che/api/project/server/ProjectService.java [hunk 1]
```diff
@@ -887,25 +887,17 @@ public ItemReference getItem(@ApiParam(value = "Workspace ID", required = true)
...
-        final List<ItemReference> items = new ArrayList<>(length);
+        final List<SearchResultEntry> searchResultEntries = result.getResults();
+        final List<ItemReference> items = new ArrayList<>(searchResultEntries.size());
         final FolderEntry root = projectManager.getProjectsRoot();
 
-        List<SearchResultEntry> entries = result.getResults();
-        for (int i = skipCount; i < length; i++) {
-            final VirtualFileEntry child = root.getChild(entries.get(i).getFilePath());
+        for (SearchResultEntry searchResultEntry : searchResultEntries) {
+            final VirtualFileEntry child = root.getChild(searchResultEntry.getFilePath());
```
Comment (skabashnyuk) [inline wsagent/che-core-api-project/src/main/java/org/eclipse/che/api/project/server/ProjectService.java:900]: Stream API?

Comment (RomanNikitenko) [inline wsagent/che-core-api-project/src/main/java/org/eclipse/che/api/project/server/ProjectService.java:900]: @skabashnyuk  I don't know how I can improve this
The reason is root.getChild() throws ServerException
So I can't use Stream API here

Comment (skabashnyuk) [inline wsagent/che-core-api-project/src/main/java/org/eclipse/che/api/project/server/ProjectService.java:900]: ok

**Comment by skabashnyuk** [comment]
ok

**Comment by vparfonov** [comment]
ok

**Pull request merged**